pkgs/misc/fastly/default.nix

```
@@ -0,0 +1,53 @@
+{ lib, fetchFromGitHub, installShellFiles, buildGoModule }:
+
+buildGoModule rec {
+  pname = "fastly";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "fastly";
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "sha256-pMe9hsjxJ4zyzupUWWU2tc5sJb9xhIIdG/Xjh2d7XtY=";
+    # The git commit is part of the `fastly version` original output;
+    # leave that output the same in nixpkgs.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD | head -c7 > $out/COMMIT
+      find "$out" -name .git -print0 | xargs -0 rm -rf
+    '';
+  };
+
+  subPackages = [ "cmd/fastly" ];
+
+  vendorSha256 = "sha256-5MvJS10f7YLvO+wCmUJleU27hCJbsNrOIfUZnniGw+E=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  # Flags as provided by the build automation of the project:
+  #   https://github.com/fastly/cli/blob/7844f9f54d56f8326962112b5534e5c40e91bf09/.goreleaser.yml#L14-L18
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/fastly/cli/pkg/revision.AppVersion=v${version}"
+    "-X github.com/fastly/cli/pkg/revision.Environment=release"
+  ];
+  preBuild = ''
+    ldflags+=" -X github.com/fastly/cli/pkg/revision.GitCommit=$(cat COMMIT)"
+    ldflags+=" -X 'github.com/fastly/cli/pkg/revision.GoVersion=$(go version)'"
+  '';
+
+  postInstall = ''
+    HOME=/tmp $out/bin/fastly --completion-script-bash > fastly.bash
+    HOME=/tmp $out/bin/fastly --completion-script-zsh > fastly.zsh
+    installShellCompletion fastly.{bash,zsh}
+  '';
```

```suggestion
  postInstall = ''
    export HOME="$(mktemp -d)"
    installShellCompletion --cmd fastly \
      --bash <($out/bin/fastly --completion-script-bash) \
      --zsh <($out/bin/fastly --completion-script-zsh)
  '';
```

it's typically a bit safer to use `mktemp -d` than using `/tmp` directly. the rest of the change is pretty optional

are the completions now working on both platforms? I think I saw a mention of some issues before

---

Changed. Yup, completions are working on both platforms with bash and zsh.